cocotb/share/lib/vpi/VpiTrampoline.cpp

```diff
@@ -0,0 +1,228 @@
+        static PLI_INT32  vpi_put_data        (PLI_INT32 id, PLI_BYTE8 *dataLoc, PLI_INT32 numOfBytes)                             { static fptr_vpi_put_data              f = reinterpret_cast<fptr_vpi_put_data>             (resolve_function("vpi_put_data"             )); return f(id, dataLoc, numOfBytes);            }
+        static void      *vpi_get_userdata    (vpiHandle obj)                                                                      { static fptr_vpi_get_userdata          f = reinterpret_cast<fptr_vpi_get_userdata>         (resolve_function("vpi_get_userdata"         )); return f(obj);                                }
+        static PLI_INT32  vpi_put_userdata    (vpiHandle obj, void *userdata)                                                      { static fptr_vpi_put_userdata          f = reinterpret_cast<fptr_vpi_put_userdata>         (resolve_function("vpi_put_userdata"         )); return f(obj, userdata);                      }
+        static PLI_INT32  vpi_vprintf         (const PLI_BYTE8 *format, va_list ap)                                                { static fptr_vpi_vprintf               f = reinterpret_cast<fptr_vpi_vprintf>              (resolve_function("vpi_vprintf"              )); return f(format, ap);                         }
+        static PLI_INT32  vpi_mcd_vprintf     (PLI_UINT32 mcd, const PLI_BYTE8 *format, va_list ap)                                { static fptr_vpi_mcd_vprintf           f = reinterpret_cast<fptr_vpi_mcd_vprintf>          (resolve_function("vpi_mcd_vprintf"          )); return f(mcd, format, ap);                    }
+        static PLI_INT32  vpi_flush           (void)                                                                               { static fptr_vpi_flush                 f = reinterpret_cast<fptr_vpi_flush>                (resolve_function("vpi_flush"                )); return f();                                   }
+        static PLI_INT32  vpi_mcd_flush       (PLI_UINT32 mcd)                                                                     { static fptr_vpi_mcd_flush             f = reinterpret_cast<fptr_vpi_mcd_flush>            (resolve_function("vpi_mcd_flush"            )); return f(mcd);                                }
+        static PLI_INT32  vpi_control         (PLI_INT32 operation, ...)                                                           { va_list ap; va_start(ap, operation); PLI_INT32 ret = vpi_vcontrol(operation, ap); va_end(ap); return ret; }
+        static vpiHandle  vpi_handle_by_multi_index (vpiHandle obj, PLI_INT32 num_index, PLI_INT32 *index_array)                   { static fptr_vpi_handle_by_multi_index f = reinterpret_cast<fptr_vpi_handle_by_multi_index>(resolve_function("vpi_handle_by_multi_index")); return f(obj, num_index, index_array);        }
+
+        static PLI_INT32 vpi_vcontrol         (PLI_INT32 operation, va_list ap)
```

Comment:
You can avoid this special case with
```suggestion
        template<typename...Args>
        static PLI_INT32 vpi_vcontrol         (PLI_INT32 operation, Args... args)
```
Then use `f(operation, args...)` later.